dashboards/resource_lakeview_dashboard.go

```diff
@@ -0,0 +1,179 @@
+package dashboards
+
+import (
+	"context"
+	"crypto/md5"
+	"fmt"
+	"log"
+
+	"github.com/databricks/databricks-sdk-go/service/dashboards"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// Calculates MD5 hash of the given content
+func calculateMd5Hash(content []byte) string {
+	return fmt.Sprintf("%x", md5.Sum(content))
+}
+
+// Reads content from a JSON string or a file path and returns the content and its MD5 hash
+func readSerializedJsonContent(jsonStr, filePath string) (serJSON string, md5Hash string, err error) {
+	var content []byte
+	if filePath != "" {
+		content, err = common.ReadFileContent(filePath)
+		if err != nil {
+			return "", "", err
+		}
+	} else {
+		log.Printf("[INFO] Reading `serialized_json` of %d bytes", len(jsonStr))
+		content = []byte(jsonStr)
+	}
+	md5Hash = calculateMd5Hash(content)
+	return string(content), md5Hash, nil
+}
+
+// ResourceLakeviewDashboard manages lakeview dashboards
+func ResourceLakeviewDashboard() common.Resource {
+	s := common.StructToSchema(dashboards.Dashboard{},
+		func(s map[string]*schema.Schema) map[string]*schema.Schema {
+			s["create_time"].Computed = true
+			s["dashboard_id"].Computed = true
+			s["display_name"].Optional = false
+			s["display_name"].Required = true
+			s["etag"].Computed = true
+			s["lifecycle_state"].Computed = true
+			s["parent_path"].Optional = false
+			s["parent_path"].Required = true
+			s["parent_path"].ForceNew = true
+			s["path"].Computed = true
+			s["serialized_dashboard"].ConflictsWith = []string{"file_path"}
+			s["serialized_dashboard"].DiffSuppressFunc = func(k, old, new string, d *schema.ResourceData) bool {
+				_, new_json_hash, err := readSerializedJsonContent(new, d.Get("file_path").(string))
+				if err != nil {
+					return false
+				}
+				return d.Get("md5").(string) == new_json_hash && !d.Get("dashboard_change_detected").(bool)
+			}
+			s["update_time"].Computed = true
+			s["warehouse_id"].Optional = false
+			s["warehouse_id"].Required = true
+
+			s["file_path"] = &schema.Schema{
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"serialized_dashboard"},
+			}
+			s["md5"] = &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
```

Comment:
I am not sure this needs to be computed. Looking at the `files` resource, it should not. Marking it as computed also supresses diffs, which can have unintended consequences.

Reply:
2 points:
1) In files, it is marked as optional. I know it is not exposed to the user but it is better that user cannot change this field (even unintentionally)
2) We do not have a Suppress on this field. We use this field indirectly in Diffsuppress for other field which fetches the 'md5' field from state so might not cause issue
But I am open to your opinion